package/src/index.ts

```diff
@@ -70,6 +114,255 @@ export class CandleClient {
+      };
+    } else if (tradeAsset.transportAsset !== undefined) {
+      return {
+        assetKind: "transport",
+        description: tradeAsset.transportAsset.description,
+        imageURL: tradeAsset.transportAsset.imageURL,
+        name: tradeAsset.transportAsset.name,
+        originAddress: {
+          value: tradeAsset.transportAsset.originAddress.value,
+        },
+        destinationAddress: {
+          value: tradeAsset.transportAsset.destinationAddress.value,
+        },
+        originCoordinates: {
+          latitude: tradeAsset.transportAsset.originCoordinates.latitude,
+          longitude: tradeAsset.transportAsset.originCoordinates.longitude,
+        },
+        destinationCoordinates: {
+          latitude: tradeAsset.transportAsset.destinationCoordinates.latitude,
+          longitude: tradeAsset.transportAsset.destinationCoordinates.longitude,
+        },
+        seats: tradeAsset.transportAsset.seats,
+        linkedAccountID: tradeAsset.transportAsset.linkedAccountID,
+        serviceAssetID: tradeAsset.transportAsset.serviceAssetID,
+        serviceTradeID: tradeAsset.transportAsset.serviceTradeID,
+      };
+    } else if (tradeAsset.otherAsset !== undefined) {
+      return { assetKind: "other" };
+    } else if (tradeAsset.nothingAsset !== undefined) {
+      return { assetKind: "nothing" };
+    } else {
+      return undefined;
+    }
+  }
+
+  private convertToCounterparty(
+    counterparty: InternalCounterparty
+  ): Counterparty {
+    if (counterparty.merchantCounterparty !== undefined) {
+      return {
+        kind: "merchant",
+        name: counterparty.merchantCounterparty.name,
+        logoURL: counterparty.merchantCounterparty.logoURL,
+        location: counterparty.merchantCounterparty.location,
+      };
+    } else if (counterparty.userCounterparty !== undefined) {
+      return {
+        kind: "user",
+        avatarURL: counterparty.userCounterparty.avatarURL,
+        legalName: counterparty.userCounterparty.legalName,
+        username: counterparty.userCounterparty.username,
+      };
+    } else if (counterparty.serviceCounterparty !== undefined) {
+      return {
+        kind: "service",
+        service: counterparty.serviceCounterparty.service,
+      };
+    } else {
+      throw new Error("Unknown counterparty kind");
+    }
+  }
+
+  private convertToAssetAccount(account: NitroAssetAccount): AssetAccount {
+    if (account.details.fiatAccountDetails !== undefined) {
+      const d = account.details.fiatAccountDetails;
+      return {
+        legalAccountKind: account.legalAccountKind,
+        nickname: account.nickname,
+        details: {
+          assetKind: "fiat",
+          serviceAccountID: d.serviceAccountID,
+          currencyCode: d.currencyCode,
+          balance: d.balance,
+          linkedAccountID: d.linkedAccountID,
+          service: d.service,
+          ach: d.ach,
+          wire: d.wire,
+        },
+      };
```

Comment:
```suggestion
      return {
        ...account,
        details: {
          ...account.details,
          assetKind: "fiat",
        },
      };
```

#pp can be much less verbose (same for all of these mapper functions)